### PR TITLE
Issue #3665: New layout templates *installed manually* require theme cache clear

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/3665

### CODE REVIEW

File: core/modules/layout/layout.admin.inc [hunk 1]
```diff
@@ -584,9 +584,14 @@ function layout_settings_form_update_layout($form, &$form_state) {
...
   }
   if (isset($form_state['values']['layout_template'])) {
-    $new = $form_state['values']['layout_template'];
-    $region_mapping = layout_get_legacy_region_mapping($layout->layout_template, $new);
-    $layout->setLayoutTemplate($form_state['values']['layout_template'], $region_mapping);
+    $old_template = $layout->layout_template;
+    $new_template = $form_state['values']['layout_template'];
+    if ($old_template != $new_template) {
+      $region_mapping = layout_get_legacy_region_mapping($old_template, $new_template);
+      $layout->setLayoutTemplate($form_state['values']['layout_template'], $region_mapping);
```
Comment (docwilmot) [inline core/modules/layout/layout.admin.inc]: why not `$layout->setLayoutTemplate($new_template, $region_mapping)`?

Comment (klonos) [reply core/modules/layout/layout.admin.inc]: Indeed, why not. Fixed1 😉

**Comment by klonos** [comment]
...the usual false-positive `fail comment-new class not found. comment.test:628`

File: core/modules/layout/layout.admin.inc [hunk 2]
```diff
@@ -589,9 +589,14 @@ function layout_settings_form_update_layout($form, &$form_state) {
...
   }
   if (isset($form_state['values']['layout_template'])) {
-    $new = $form_state['values']['layout_template'];
-    $region_mapping = layout_get_legacy_region_mapping($layout->layout_template, $new);
-    $layout->setLayoutTemplate($form_state['values']['layout_template'], $region_mapping);
+    $old_template = $layout->layout_template;
+    $new_template = $form_state['values']['layout_template'];
+    if ($old_template != $new_template) {
+      $region_mapping = layout_get_legacy_region_mapping($old_template, $new_template);
+      $layout->setLayoutTemplate($new_template, $region_mapping);
```
Comment (docwilmot) [inline core/modules/layout/layout.admin.inc:596]: Perhaps, as a compromise, instead we could simply set a message here saying Backdrop detected a template change, please flash your theme cache if you just added a new template and your regions are not showing what you expect. Personally I prefer to flush caches automatically. 80% would UX benefit surely.

Comment (docwilmot) [reply core/modules/layout/layout.admin.inc:596]: Or better still, is there any way to check if there is a theme entry for the new template and then we'd know for sure that this is the issue, then just explicitly message "you need to flush your theme cache" in nicer words of course.

**Comment by klonos** [comment]
Closing in favor of the merged https://github.com/backdrop/backdrop/pull/3107

**Pull request closed without merging**